File: backend/app/schemas/task.py
from datetime import datetime

from pydantic import BaseModel


class IngestionTaskResponse(BaseModel):
    id: int
    document_id: int
    status: str
    progress: int
    message: str | None
    started_at: datetime | None
    finished_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/task.ts
import http from "./http";

export interface IngestionTaskItem {
  id: number;
  document_id: number;
  status: string;
  progress: number;
  message: string | null;
  started_at: string | null;
  finished_at: string | null;
  created_at: string;
}

export async function getTasks(documentId?: number) {
  const { data } = await http.get<IngestionTaskItem[]>("/tasks", {
    params: documentId ? { document_id: documentId } : undefined
  });
  return data;
}
